components/scream/scripts/scripts-tests

```diff
@@ -484,8 +500,19 @@ class TestTestAllScream(TestBaseOuter.TestBase):
 
     def test_mpirank_and_thread_spreading(self):
         """
-        Test that MPI ranks and threads are properly spread across a machine's computing resources (TODO)
+        Test that MPI ranks and threads are properly spread across a machine's computing resources
         """
+        if self._full and self._machine == "mappy":
+            cmd = self.get_cmd("./test-all-scream -x -e OMP_DISPLAY_ENV=True -e SCREAM_TEST_THREAD_SPREAD=True -m $machine -b HEAD -k -t dbg --ctest-parallel-level=40 ", self._machine, dry_run=False)
+            output = run_cmd_assert_result(self, cmd, from_dir=TEST_DIR)
+            test_omp_spread(self, output, 40)
+
+            cmd = self.get_cmd("./test-all-scream -x -e OMP_DISPLAY_ENV=True -e SCREAM_TEST_RANK_SPREAD=True -m $machine -b HEAD -k -t dbg --ctest-parallel-level=40 ", self._machine, dry_run=False)
```

Comment:
Why the `-k` flag?

Reply:
Ah, I didn't realize we changed the repo during the scripts test.

Reply:
@bartgol , we don't. But it is possible the user could have a dirty repo when they run `scripts-tests`